base/strings/search.jl

```diff
@@ -166,17 +185,22 @@ function _search_bloom_mask(c)
 end
 
 _nthbyte(s::String, i) = codeunit(s, i)
-_nthbyte(a::Union{AbstractVector{UInt8},AbstractVector{Int8}}, i) = a[i]
+_nthbyte(t::AbstractVector, index) = t[index + (firstindex(t)-1)]
 
 function _searchindex(s::String, t::String, i::Integer)
     # Check for fast case of a single byte
     lastindex(t) == 1 && return something(findnext(isequal(t[1]), s, i), 0)
     _searchindex(unsafe_wrap(Vector{UInt8},s), unsafe_wrap(Vector{UInt8},t), i)
 end
 
-function _searchindex(s::ByteArray, t::ByteArray, i::Integer)
-    n = sizeof(t)
-    m = sizeof(s)
+function _searchindex(s::AbstractVector{<:Union{Int8,UInt8}},
+                      t::AbstractVector{<:Union{Int8,UInt8}},
+                      _i::Integer)
+    require_one_based_indexing(s)
```

Comment:
Why do we need to require this, since `_nthbyte` handles non 1-based indices?

Reply:
Aren't you testing with an offset array?  Why didn't it throw an error here?

Reply:
Because I wasn't testing offset array that has `firstindex` < 0; the reason this won't work is: https://github.com/JuliaLang/julia/blob/0e082ee1f173bb5992e02496b2de12e5ec98a2e5/base/strings/search.jl#L244 . In short, we can't tell the difference between "not found" (return `0`) and "found in offset array with idx `0`".

Reply:
We should definitely support an arbitrary `firstindex` here.   We could just return `typemin(Int)` as the sentinel value.